lib/ClangImporter/ImportMacro.cpp

```diff
@@ -334,6 +345,29 @@ static ValueDecl *importMacro(ClangImporter::Implementation &impl,
         tokenI[3].is(clang::tok::r_paren)) {
       return importStringLiteral(impl, DC, macro, name, tokenI[2],
                                  MappedStringLiteralKind::CFString, ClangN);
+    // Check for literals with a cast, e.g. "(cpu_type_t) 1"
+    } else if (tokenI[0].is(clang::tok::l_paren) &&
+               tokenI[1].is(clang::tok::identifier) &&
```

Comment:
I thought about parsing the cast before the switch and handling it in a similar way to the parentheses, i.e. dropping the tokens comprising the cast from the list. It certainly is more generic in a way that the code for importing literals with or without sign tokens doesn't need to be repeated. However, I have a feeling that this is not safe, for example for recursively defined macros

```
#define FOO_1 (char) 1024
#define FOO_2 (int) FOO_1
```

In that case, we would import `FOO_2` as an integer, skipping the intermediate cast to char.